src/timesync/timesyncd-bus.c

```diff
@@ -23,24 +42,86 @@ static int property_get_servers(
                 void *userdata,
                 sd_bus_error *error) {
 
-        ServerName *p, **s = userdata;
-        int r;
+        ServerName **s = userdata;
 
         assert(s);
         assert(bus);
         assert(reply);
 
-        r = sd_bus_message_open_container(reply, 'a', "s");
+        return reply_server_names(*s, reply);
+}
+
+static int property_get_runtime_servers(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *reply,
+                void *userdata,
+                sd_bus_error *error) {
+
+        Manager *m = userdata;
+
+        assert(m);
+
+        return reply_server_names(m->runtime_servers, reply);
+}
+
+static int property_set_runtime_servers(
+                sd_bus *bus,
+                const char *path,
+                const char *interface,
+                const char *property,
+                sd_bus_message *message,
+                void *userdata,
+                sd_bus_error *error) {
+
+        _cleanup_strv_free_ char **names = NULL;
+        Manager *m = userdata;
+        char **name;
+        int r;
+
+        assert(m);
+        assert(bus);
+        assert(message);
+
```

Comment:
so, writable D-Bus properties kinda suck, except for stuff tht really is variable like and isolated from everything else. I doubt this one qualifies. hence, please just make the setter a regular method (i.e. SetDynamicServer()).

Also, please hook up with PK. See hwo bus_link_method_set_dns_servers_internal() does that in resolved.

Reply:
Hmm, could you elaborate on the reason for this? Properties give you signals on change for free as well, and my understanding of them is that they are meant for exactly the use-case at hand: expose an internal setting to the outside world.

Reply:
read-only properties are OK and we use them all over the place. My issue is with writable properties. for example you cannot do PolicyKit (or any other form of "slow" auth) with them, and we usually talk about system services here...

I am not asking to remove the property, i am just saying: make it read-only and instead provide a setter method, that can do polkit can report proper errors, and so on.

read-only props doesn't mean the props can't change value. It just means you cannot write to them using the generic propery set APIs. And yes, you can still fire PropertiesChanged signals as you wish also for read-only props.

Reply:
or to say this differently: i think adding a generic concept of writing to props to the dbus props spec was a mistake. Generalizing reading props, and generalizing props changes is a good thing, but generalizing writing to props was a mistake. It might make sense on the user bus, where access control doesn't matter (but maybe not even then anymore, given flatpak and app sandboxing) but our stuff is almost entirely system level stuff, and thus should not really use writable props.

Reply:
Okay, understood. Implemented as requested now.